js/ui/select_box.js

```diff
@@ -109,7 +110,11 @@ var SelectBox = DropDownList.inherit({
                 this._cancelEditing();
             },
             enter: function(e) {
-                if(this._input().val() === "" && this.option("value") && this.option("allowClearing")) {
+                var inputText = this._input().val();
```

Comment:
`this._input().val().trim()`

Reply:
If customer types " ", it will be added to list. Is it OK?